resources/js/views/RegisterPage.vue

```diff
@@ -0,0 +1,128 @@
+					<InputText id="email" v-model="email" autocomplete="email" />
+					<label for="email">{{ $t("profile.login.email") }}</label>
+				</FloatLabel>
+				<FloatLabel variant="on">
+					<InputPassword id="password" v-model="password" autocomplete="new-password" />
+					<label for="password">{{ $t("profile.login.new_password") }}</label>
+				</FloatLabel>
+				<FloatLabel variant="on">
+					<InputPassword id="password_confirmation" v-model="passwordConfirmation" autocomplete="new-password" />
+					<label for="password_confirmation">{{ $t("profile.login.confirm_new_password") }}</label>
+				</FloatLabel>
+				<Message v-if="confirmationError" severity="error" class="text-sm mt-2">
+					{{ confirmationError }}
+				</Message>
+			</div>
+			<div class="flex items-center mt-9">
+				<Button @click="register" :disabled="!isFormValid" severity="contrast" class="w-full font-bold border-none rounded-xl">
+					{{ $t("profile.register.signup") }}
+				</Button>
+			</div>
+		</form>
+	</Panel>
+</template>
+
+<script setup lang="ts">
+import FloatLabel from "primevue/floatlabel";
+import Button from "primevue/button";
+import InputText from "@/components/forms/basic/InputText.vue";
+import InputPassword from "@/components/forms/basic/InputPassword.vue";
+import InitService from "@/services/init-service";
+import ProfileService from "@/services/profile-service";
+import { useLeftMenuStateStore } from "@/stores/LeftMenuState";
+import { useLycheeStateStore } from "@/stores/LycheeState";
+import { storeToRefs } from "pinia";
+import Panel from "primevue/panel";
+import { ref, computed, onMounted } from "vue";
+import { useRouter } from "vue-router";
+import { trans } from "laravel-vue-i18n";
+import { useToast } from "primevue/usetoast";
+import Message from "primevue/message";
+
+const router = useRouter();
+const lycheeStore = useLycheeStateStore();
+const leftMenuStore = useLeftMenuStateStore();
+const { title } = storeToRefs(lycheeStore);
+const is_loaded = ref(false);
+
+const username = ref("");
+const email = ref("");
+const password = ref("");
+const passwordConfirmation = ref("");
+const errorMessage = ref("");
+
+const confirmationError = computed(() => {
+	return password.value !== passwordConfirmation.value ? trans("profile.register.password_mismatch") : "";
+});
+
+const isFormValid = computed(() => {
+	return username.value && email.value && password.value && passwordConfirmation.value && !confirmationError.value;
+});
+
+function goBack() {
+	router.push({ name: "gallery" });
+}
+
+const toast = useToast();
+
+function register() {
+	ProfileService.register({
+		username: username.value,
+		email: email.value,
+		password: password.value,
+		password_confirmation: passwordConfirmation.value,
+	})
+		.then(() => {
+			errorMessage.value = ""; // Clear error message on success
+			toast.add({ severity: "success", summary: trans("profile.register.success"), life: 3000 });
+			router.push({ name: "gallery" }); // Redirect to gallery
```

Comment:
Pushing to gallery here without a refresh still shows sign up / sign in buttons. 

Is there a way to trigger a redirect and a refresh?

Reply:
I'll dig into that.

Reply:
that is weird, because server side we do the login.